homeassistant/components/media_player/cambridgeaudio.py

```diff
@@ -0,0 +1,389 @@
+"""
+Support for Cambridge Audio Network Audio Players (StreamMagic platform).
+
+Example configuration.yaml entry:
+media_player:
+    platform: cambridgeaudio
+    host: 192.168.x.y
+    command_off: "OFF"
+    name: "CA 851N"
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.cambridgeaudio/
+"""
+import logging
+# pylint: disable=unused-import
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    SUPPORT_TURN_OFF, SUPPORT_TURN_ON,
+    SUPPORT_PAUSE, SUPPORT_PLAY, SUPPORT_PREVIOUS_TRACK, SUPPORT_NEXT_TRACK,
+    SUPPORT_STOP, SUPPORT_SEEK, SUPPORT_SHUFFLE_SET,
+    SUPPORT_SELECT_SOURCE, SUPPORT_CLEAR_PLAYLIST,
+    SUPPORT_VOLUME_SET, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_STEP,
+    MediaPlayerDevice, PLATFORM_SCHEMA, MEDIA_TYPE_MUSIC)
+
+from homeassistant.const import (STATE_ON, STATE_OFF, STATE_UNKNOWN,
+                                 STATE_PLAYING, STATE_PAUSED, STATE_IDLE,
+                                 CONF_HOST, CONF_NAME, CONF_COMMAND_OFF)
+
+import homeassistant.helpers.config_validation as cv
+import homeassistant.util.dt as dt_util
+
+REQUIREMENTS = ['stream_magic==0.16']
+DOMAIN = 'cambridgeaudio'
+_LOGGER = logging.getLogger(__name__)
+
+CONF_SOURCES = 'sources'
+DEFAULT_NAME = 'Cambridge Audio Streamer'
+DEFAULT_PWROFF_CMD = 'OFF'
+KNOWN_HOSTS = []
```

Comment:
Do not use a global variable. Instead use `hass.data`. This is a dictionary that can store global info. Use a key that contains the platform name.

Reply:
```python
DATA_KEY = 'cambrdigeaudio'
hass.data[DATA_KEY] = []
```

Reply:
Ok, I just changed this to be hardcoded. It doesn't really make sense to have this configurable in the platform and the config file.